src/main/webapp/app/exercises/modeling/manage/modeling-exercise-detail.component.html

```diff
@@ -6,12 +6,22 @@ <h2><span jhiTranslate="artemisApp.modelingExercise.detail.title">Modeling Exerc
             <jhi-alert></jhi-alert>
             <jhi-alert-error></jhi-alert-error>
             <div class="text-start">
-                <jhi-non-programming-exercise-detail-common-actions
-                    *ngIf="modelingExercise.course"
-                    [exercise]="modelingExercise"
-                    [courseId]="modelingExercise.course.id!"
-                    [isExamExercise]="false"
-                ></jhi-non-programming-exercise-detail-common-actions>
+                <div *ngIf="!isExamExercise">
+                    <jhi-non-programming-exercise-detail-common-actions
+                        *ngIf="modelingExercise.course"
+                        [isExamExercise]="isExamExercise"
+                        [courseId]="modelingExercise.course.id!"
+                        [exercise]="modelingExercise"
+                    ></jhi-non-programming-exercise-detail-common-actions>
+                </div>
+                <div *ngIf="isExamExercise">
+                    <jhi-non-programming-exercise-detail-common-actions
+                        *ngIf="modelingExercise.exerciseGroup?.exam?.course"
+                        [isExamExercise]="isExamExercise"
+                        [courseId]="modelingExercise.exerciseGroup!.exam!.course!.id!"
+                        [exercise]="modelingExercise"
+                    ></jhi-non-programming-exercise-detail-common-actions>
+                </div>
```

Comment:
a thought about this if/else approach. One could also declare the course in the respective component like this:
```typescript
this.course = isExamExercise ? modelingExercise.exerciseGroup?.exam?.course : modelingExercise.course;
```
and use it as input for the component. This would simplify the html code to:
```html
 <jhi-non-programming-exercise-detail-common-actions
      *ngIf="course"
      [isExamExercise]="isExamExercise"
      [courseId]="course!.id!"
      [exercise]="modelingExercise"
  ></jhi-non-programming-exercise-detail-common-actions>
```

Reply:
This indeed removes some duplication and makes the code more understandable in my opinion!